File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/C++/s867221485_commented.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int N;
    cin >> N;
    cin.ignore();
    string s;
    getline(cin, s);

    int ans = 0;
    int sha = 0;
    int dot = 0;

    for (char c : s) {
        if (c == '#') {
            if (sha <= dot) {
                ans += sha;
                sha = 0;
                dot = 0;
            }
            sha++;
        } else {
            dot++;
        }
    }

    if (sha <= dot) {
        ans += sha;
        dot = 0;
    }

    cout << ans + dot << endl;
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s867221485_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	line := scanner.Text()
	N, _ := strconv.Atoi(line)
	scanner.Scan()
	s := scanner.Text()

	flag := false
	ans := 0
	hen := 0
	sha := 0
	dot := 0
	temp := 0
	first := true

	for i := 0; i < len(s); i++ {
		c := s[i]
		if c == '#' {
			first = false
			if sha <= dot {
				ans += sha
				sha = 0
				dot = 0
				first = true
			} else {
				// temp += dot
			}
			sha++
		} else {
			dot++
		}
	}

	if sha <= dot {
		ans += sha
		dot = 0
	}

	fmt.Println(ans + dot)
}

// End of Code
